### PR TITLE
Fix stacking of rockets

### PR DESCRIPTION
**Pull request by teccheck**
Description: See #289

I'm open to any suggestions for this change

### CODE REVIEW

**Comment by enteryournamehere** [comment]
@MickVermeulen or @Wincent01 I would like to hear your thoughts on this, see also the comments in #289. Is this vehicle-stack-size-1 check correct, or should stacking be prevented by items having different subkeys?

**Comment by MickVermeulen** [comment]
I think the best bet would be to uniquely store any item that has LDF data (which rockets and race cars do) as otherwise it's possible to overwrite this data. So I'd implement this to check whether or not there's LDF and refuse stacking if so.

**Comment by teccheck** [comment]
So you mean checking if `ItemComponent.Settings` is not null in which case the method would just skip the stacking part?

**Comment by MickVermeulen** [comment]
Just alter your current commit to check for the settings instead of checking for the item type. Make sure to test it, though :P

**Comment by teccheck** [comment]
Should I add another commit to this pull request or should I create a new one without the first commit?

**Comment by MickVermeulen** [comment]
Add another :)

**Comment by teccheck** [comment]
Hi, I don't want to annoy anyone... But is there anything left to do on this pull request?

**Comment by enteryournamehere** [comment]
If it's ready now, I can test it tomorrow to make sure it works well, after that's done it can be merged

**Comment by teccheck** [comment]
I'd say it's ready

File: Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs [hunk 1]
```diff
@@ -363,6 +363,9 @@ private static async Task<ItemComponent> GetItemComponentForLotAysnc(Lot lot)
             if (stackSize == default) 
                 stackSize = int.MaxValue;
             
+            if(settings != null)
```
Comment (enteryournamehere) [inline Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs]: Small [style](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) comment: there should be a space between `if` and the parentheses

**Review by enteryournamehere — COMMENTED** [review]
Tested it, it works well :ok_hand:
Just one small comment, after that's updated it can be merged (unless @TheNexusAvenger has any other comments).

File: Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs (hunk 1 above)
Comment (TheNexusAvenger) [reply Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs]: The change seems small and simple, and you tested it. I don't have anything else to comment. Just that styling comment you left.

**Comment by teccheck** [comment]
Ok, done :)

**Comment by enteryournamehere** [comment]
Perfect thanks!

**Pull request merged**